src/main/java/uk/gov/hmcts/ccd/service/MessagePublisherRunnable.java

```diff
@@ -63,10 +65,31 @@ private void processUnpublishedMessages(Pageable pageable,
         }
     }
 
+    private String getProperty(JsonNode data, String header) {
+        try {
+            return data.get(header).asText();
+        } catch (NullPointerException e) {
+            return "null";
+        }
+    }
+
     private void publishMessages(Slice<MessageQueueCandidateEntity> messagesToPublish,
                                  List<MessageQueueCandidateEntity> processedEntities) {
         messagesToPublish.get().forEach(entity -> {
-            jmsTemplate.convertAndSend(publishMessageTask.getDestination(), entity.getMessageInformation());
+            jmsTemplate.convertAndSend(
+                publishMessageTask.getDestination(),
+                entity.getMessageInformation(),
+                messagePostProcessor -> {
+                    for (String property : properties) {
+                        if (!(getProperty(entity.getMessageInformation(), property).equals("null"))) {
```

Comment:
why is this check required? If we are building the message to be sent here, how is it possible that the message already contains the property?